website/docs/tutorials/07-testing/README.md

```diff
@@ -140,15 +138,75 @@ func test_tested_property()
 
 The setup case hook is bound to a matching test case and is executed just before the test case
 itself.
-Use case hooks to configure the behavior of particular test case,
-for example, by calling the [`max_examples`](./02-cheatcodes/max-examples.md) cheatcode.
+The hook is executed within a context built by the `__setup__` hook,
+but in isolation for each test separately.
+Then, Protostar immediately executes the test case function.
+This makes them useful to extract test-specific setup logic from tested code itself.
 
-```cairo title="Using setup case hook"
-%lang starknet
+```cairo title="Using setup case hook to prepare test-specific state"
+@external
+func __setup__{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    range_check_ptr
+}() {
+    balance.write(10);
+    return ();
+}
 
+@external
+func setup_need_more_money{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    range_check_ptr
+}() {
+    balance.write(10000);
```

Comment:
Should we add balance definition to the code? Somebody may get confused

Reply:
I have thought about doing so, but decided not to because this example grew to enormous size and is occupying a lot of eye space. In the same manner, I have removed imports and `%lang starknet` lines from example.

Reply:
We will add it if somebody asks about it